internal/bootloader/grub.go

```diff
@@ -0,0 +1,73 @@
+//go:build termus
+// +build termus
+
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package bootloader
+
+import (
+	"path/filepath"
+
+	"github.com/canonical/pebble/internal/bootloader/grubenv"
+	"github.com/canonical/pebble/internal/osutil"
+)
+
+// grub implements the Bootloader interface to support bootloader operations
+// on GRUB 2.
+type grub struct {
+	rootdir string
+	env     *grubenv.Env
+}
+
+// newGrub initializes a new instance of the GRUB bootloader.
+func newGrub(rootdir string) Bootloader {
+	return &grub{
+		rootdir: rootdir,
+		env:     grubenv.NewEnv(filepath.Join(rootdir, "grubenv")),
```

Comment:
spec changed to grub.env if that is still on.

Reply:
Nice catch! Thank you